### PR TITLE
Mock Schema Registry 

### PR DESCRIPTION
**Pull request by chengjie8**
## What was the problem?
There's no mock schema registry in place to conduct local testing using avro schema serialization/deserialization 

Associated tickets or Slack threads:
- #1902 

## How does this fix it?[^1]
A new schema registry image, `confluentinc/cp-schema-registry`, has been added to serve as a mock schema registry. It will automatically register five BIE contention classification schemas once the service is up.

## How to test this PR
run e2e integration test

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Review by msnwatson — CHANGES_REQUESTED** [review]
Please be sure to add to the dependabot.yml file with the location of this new Dockerfile because there is a limitation where dependabot cannot recursively search for dockerfiles to keep track of updates.

Also, since we are adding a new container, it might be worth linking a successful SecRel run to be sure we are not introducing any new vulnerabilities from the base image.

**Comment by chengjie8** [comment]
> Please be sure to add to the dependabot.yml file with the location of this new Dockerfile because there is a limitation where dependabot cannot recursively search for dockerfiles to keep track of updates.
> 
> Also, since we are adding a new container, it might be worth linking a successful SecRel run to be sure we are not introducing any new vulnerabilities from the base image.

Sounds good. I will do that.

**Comment by chengjie8** [comment]
[latest scan has been completed here](https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/runs/6395504338) did not bring up additional issues. @msnwatson

File: mocks/mock-bie-schema-registry/build.gradle [hunk 1]
```diff
@@ -0,0 +1,7 @@
+plugins {
+  id 'shared.docker.container-conventions'
+}
+
+docker {
+  // copySpec.from("src/main/resources").into(".")
```
Comment (tejans24) [inline mocks/mock-bie-schema-registry/build.gradle:6]: commented code here

File: svc-bie-kafka/src/main/resources/application-integration-test.yaml [hunk 2]
```diff
@@ -12,20 +12,20 @@ spring:
...
+#        trust-store-type: "PKCS12"
+#    admin:
+#      security.protocol: SSL
+#      ssl:
+#        keystore-location: "file:${KEYSTORE_FILE}"
+#        keystore-password: "${BIE_KAFKA_KEYSTORE_PASSWORD}"
+#        key-store-type: "PKCS12"
+#        truststore-location: "file:${TRUSTSTORE_FILE}"
+#        truststore-password: "${BIE_KAFKA_TRUSTSTORE_PASSWORD}"
+#        trust-store-type: "PKCS12"
```
Comment (tejans24) [inline svc-bie-kafka/src/main/resources/application-integration-test.yaml:31]: should these lines be removed?

**Review by tejans24 — COMMENTED** [review]
Do the current Kafka tests needed to be updated to utilize the Mock schema registry?

**Review by msnwatson — APPROVED** [review]

File: svc-bie-kafka/src/main/resources/application-integration-test.yaml (hunk 2 above)
Comment (chengjie8) [reply svc-bie-kafka/src/main/resources/application-integration-test.yaml]: reverted back, it was for testing

**Review by tejans24 — APPROVED** [review]
LGTM

**Pull request merged**